### PR TITLE
Update interactive display of text candidates

### PR DESCRIPTION
**Pull request by klshuster**
When using the tfidf retriever, makes the returned candidates look good when in interactive mode. See picture: 

screenshot 2018-03-15 16 57 23

Also, fixes issue where wikipedia task displayed the title twice in the text for the Full teacher

### CODE REVIEW

**Comment by klshuster** [comment]
Below are some examples for other datasets

personachat: 
screenshot 2018-03-16 12 27 38

AQUA:
screenshot 2018-03-16 12 27 05

convai chitchat:
screenshot 2018-03-16 12 37 04

File: parlai/tasks/wikipedia/agents.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def setup_data(self, path):
                         if self.key_value:
                             yield (title, [text]), True
                         else:
-                            yield (title + '\n' + text, None), True
+                            yield (text, None), True
```
Comment (alexholdenmiller) [inline parlai/tasks/wikipedia/agents.py]: no more title?

**Comment by jaseweston** [comment]
i'm confused, text candidates don't always have scores, and i dont think we
always want this more complex display for them
(for retrieval model yes, but text candidates isnt just for that)

File: parlai/tasks/wikipedia/agents.py (hunk 1 above)
Comment (klshuster) [reply parlai/tasks/wikipedia/agents.py]: The title is actually in the `text` variable; this was adding the title twice to the example

**Comment by klshuster** [comment]
The retrieval model generates candidate scores, though I've written the code to account for cases where the scores are not returned (and it just displays them in the table as above - see screenshot below). If you would like, I can make a cmd line arg for `interactive.py` that allows you to specify whether you would like the `text_candidates` to be displayed

screenshot 2018-03-16 18 16 26

**Comment by klshuster** [comment]
Sure, happy to discuss more in person

**Comment by klshuster** [comment]
I've updated this to add a command line argument to scripts/interactive.py, ``--prettify``, that controls whether to show the text candidates in a prettytable or not

File: parlai/scripts/interactive.py [hunk 2]
```diff
@@ -24,6 +24,9 @@ def setup_args(parser=None):
     if parser is None:
         parser = ParlaiParser(True, True)
     parser.add_argument('-d', '--display-examples', type='bool', default=False)
+    parser.add_argument('--prettify', type='bool', default=False,
```
Comment (jaseweston) [inline parlai/scripts/interactive.py]: can we call it display-prettify?

File: parlai/core/worlds.py [hunk 3]
```diff
@@ -106,6 +109,34 @@ def display_messages(msgs):
                         '|'.join(display_cands),
                         '| ...and {} more'.format(cand_len - 5)
                         )))
+        if msg.get('text_candidates'):
```
[the remark this answers is not in the document]
Comment (jaseweston) [reply parlai/core/worlds.py]: should we move this whole func to utils?

**Pull request merged**